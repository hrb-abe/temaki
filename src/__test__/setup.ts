import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as ContextHook from 'babel-plugin-require-context-hook/register'

Enzyme.configure({ adapter: new Adapter() })
ContextHook()
