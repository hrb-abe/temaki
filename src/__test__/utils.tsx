import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import { defaultTheme, ThemeProvider } from '~/modules/theme'

// 推論で型エラーがでるので、型定義してキャストする
type ShallowWithTheme = (tree: React.ReactElement) => ShallowWrapper
export const shallowWithTheme: ShallowWithTheme = tree => {
  return shallow(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}

// 推論で型エラーがでるので、型定義してキャストする
type MountWithTheme = (tree: React.ReactElement) => ReactWrapper
export const mountWithTheme: MountWithTheme = tree => {
  return mount(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}
