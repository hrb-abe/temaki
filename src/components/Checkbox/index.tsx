import * as React from 'react'
import styled from '~/modules/theme'

import * as IconFiles from '~/basics/Icons/lib/iconFiles'
import Icon from '~/components/Icon'

/**
 * Component
 */

type Props = {
    text?: string
    onClick?: (e: React.MouseEvent<HTMLDivElement>) => void
    checked?: boolean
    indeterminate?: boolean
}
const Checkbox = React.memo<Props>(
    ({ text, indeterminate, checked, ...props }) => {
        // ON の方を優先にする
        if (checked) {
            return (
                <Outer tabIndex={1} data-test="checked-box" {...props}>
                    <CheckIcon svg={IconFiles.icons.CheckBoxOn} size="24px" />
                    <Text>{text}</Text>
                </Outer>
            )
        }

        if (indeterminate) {
            return (
                <Outer tabIndex={1} data-test="indeterminate-box" {...props}>
                    <CheckIcon svg={IconFiles.icons.CheckBoxAll} size="24px" />
                    <Text>{text}</Text>
                </Outer>
            )
        }

        return (
            <Outer tabIndex={1} data-test="nocheck-box" {...props}>
                <CheckIcon svg={IconFiles.icons.CheckBoxOff} size="24px" />
                <Text>{text}</Text>
            </Outer>
        )
    }
)

/**
 * Styles
 */

const Outer = styled.div`
    display: inline-flex;
    align-items: center;
    cursor: pointer;
`

const CheckIcon = styled(Icon)`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 24px;
    width: 24px;

    & .primary {
        fill: ${props => props.theme.colors.primary.default};
    }
`

const Text = styled.span`
    height: 100%;
    font-size: 14px;
    padding-left: 4px;
    color: ${props => props.theme.colors.text};
`

export default Checkbox
