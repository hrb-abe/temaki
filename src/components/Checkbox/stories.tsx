import { action } from '@storybook/addon-actions'
import { boolean, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import Checkbox from './index'

storiesOf('Components|Checkbox', module).add('all', () => {
    return (
        <Checkbox
            onClick={action('onClick')}
            checked={boolean('Checked', false)}
            indeterminate={boolean('Indeterminate', false)}
            text={text('Text', 'Checkbox')}
        />
    )
})
