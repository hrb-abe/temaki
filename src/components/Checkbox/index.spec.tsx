import * as Enzyme from 'enzyme'
import * as React from 'react'
import { act } from 'react-dom/test-utils'
import { mountWithTheme } from '~/__test__/utils'
import Checkbox from './index'

describe('Checkbox', () => {
  let wrapper: Enzyme.ReactWrapper
  let mockOnClick: jest.Mock

  beforeEach(() => {
    mockOnClick = jest.fn()
    act(() => {
      wrapper = mountWithTheme(<Checkbox onClick={mockOnClick} />)
    })
  })

  it('should be defined', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('should match previous snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
