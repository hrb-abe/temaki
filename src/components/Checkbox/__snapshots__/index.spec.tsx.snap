// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match previous snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "grayScale": Object {
          "S0": "rgb(255, 255, 255)",
          "S10": "rgb(234, 234, 234)",
          "S100": "rgb(51, 51, 51)",
          "S20": "rgb(214, 214, 214)",
          "S30": "rgb(193, 193, 193)",
          "S40": "rgb(173, 173, 173)",
          "S5": "rgb(245, 245, 245)",
          "S50": "rgb(153, 153, 153)",
          "S60": "rgb(133, 133, 133)",
          "S70": "rgb(112, 112, 112)",
          "S80": "rgb(92, 92, 92)",
          "S90": "rgb(71, 71, 71)",
        },
        "primary": Object {
          "N20": "rgb(92, 92, 92)",
          "N40": "rgb(133, 133, 133)",
          "N60": "rgb(173, 173, 173)",
          "N80": "rgb(214, 214, 214)",
          "N95": "rgb(245, 245, 245)",
          "P20": "rgb(41, 41, 41)",
          "P40": "rgb(31, 31, 31)",
          "P60": "rgb(20, 20, 20)",
          "P80": "rgb(10, 10, 10)",
          "P95": "rgb(0, 0, 0)",
          "default": "rgb(51, 51, 51)",
        },
        "text": Object {
          "default": "#333",
        },
        "utilities": Object {
          "blue": "rgb(48, 120, 191)",
          "highlightGreen": "rgb(114, 206, 92)",
          "paleBlue": "rgb(245, 249, 252)",
          "paleGreen": "rgb(248 ,252, 247)",
          "paleRed": "rgb(253, 246, 246)",
          "red": "rgb(224, 85, 72)",
        },
      },
      "shadows": Object {
        "L1": "0 0 2px 0 rgba(0, 0, 0, 0.16)",
        "L2": "0 1px 3px 0 rgba(0, 0, 0, 0.16)",
        "L3": "0 2px 5px 0 rgba(0, 0, 0, 0.16)",
        "L4": "0 3px 8px 0 rgba(0, 0, 0, 0.16)",
        "L5": "0 5px 11px 0 rgba(0, 0, 0, 0.16)",
      },
      "typography": Object {
        "pc": Object {
          "body": Object {
            "L": "16px",
            "S": "12px",
            "XL": "18px",
            "default": "14px",
          },
          "heading": Object {
            "H1": "24px",
            "H2": "22px",
            "H3": "20px",
            "H4": "18px",
            "H5": "16px",
            "H6": "14px",
          },
        },
        "sp": Object {
          "body": Object {
            "L": "16px",
            "S": "12px",
            "XL": "18px",
            "default": "14px",
          },
          "heading": Object {
            "H1": "24px",
            "H2": "22px",
            "H3": "20px",
            "H4": "18px",
            "H5": "16px",
            "H6": "14px",
          },
        },
      },
    }
  }
>
  <Memo()
    onClick={[MockFunction]}
  >
    <styled.div
      data-test="nocheck-box"
      onClick={[MockFunction]}
      tabIndex={1}
    >
      <StyledComponent
        data-test="nocheck-box"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "gkcNzX",
              "rules": Array [
                "
  display: inline-flex;
  align-items: center;
  cursor: pointer;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[MockFunction]}
        tabIndex={1}
      >
        <div
          className="sc-bdVaJa gkcNzX"
          data-test="nocheck-box"
          onClick={[MockFunction]}
          tabIndex={1}
        >
          <styled.div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          >
            <StyledComponent
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "cGYhMO",
                    "rules": Array [
                      "
  display: flex;
  align-items: center;
  justify-content: center;
  height: 24px;
  width: 24px;

  & .primary {
    fill: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bwzfXH cGYhMO"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
              />
            </StyledComponent>
          </styled.div>
          <styled.span>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "lnTVXj",
                    "rules": Array [
                      "
  height: 100%;
  font-size: 14px;
  padding-left: 4px;
  color: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="sc-htpNat lnTVXj"
              />
            </StyledComponent>
          </styled.span>
        </div>
      </StyledComponent>
    </styled.div>
  </Memo()>
</ThemeProvider>
`;
